### PR TITLE
"Feature envy" added (Anti-pattern documentation)

### PR DESCRIPTION
**Pull request by lucasvegi**

### CODE REVIEW

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 1]
```diff
@@ -207,7 +207,83 @@ end
...
+      total - total * discount
+    else
+      total
+    end
+  end
+end
+```
+
+```elixir
+defmodule OrderItem do
```
[the remark this answers is not in the document]
Comment (lucasvegi) [reply lib/elixir/pages/anti-patterns/design-anti-patterns.md:235]: Ok. I agree with removing `OrderItem` in the example. We can leave it appearing only in the refactored code (as it is essential there)

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 2]
```diff
@@ -207,7 +207,77 @@ end
 
 ## Feature envy
 
-TODO
+#### Problem
+
+This anti-pattern occurs when a function accesses more data or calls more functions from another `Module` than from its own. The presence of this anti-pattern can make a `Module` less cohesive and increase code coupling.
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:212]:
```suggestion
This anti-pattern occurs when a function accesses more data or calls more functions from another module than from its own. The presence of this anti-pattern can make a module less cohesive and increase code coupling.
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 3]
```diff
@@ -207,7 +207,77 @@ end
...
+  def find_item(id) do
+    ...
+    %{price: ..., taxes: ..., amount: ...}
+  end
+
+  def find_discount(item) do
+    ...
+  end
+end
+```
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:245]:
```suggestion
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 4]
```diff
@@ -207,7 +207,77 @@ end
...
+    total = (item.price + item.taxes) * item.amount
+  if discount = OrderItem.find_discount(item) do
+    total - total * discount
+  else
+    total
+  end
+  end
+end
+```
+
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:233]:
```suggestion
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 5]
```diff
@@ -207,7 +207,77 @@ end
...
+  # Some functions...
+
+  def calculate_total_item(id) do
+    item = OrderItem.find_item(id)
+    total = (item.price + item.taxes) * item.amount
+  if discount = OrderItem.find_discount(item) do
+    total - total * discount
+  else
+    total
+  end
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:229]:
```suggestion

    if discount = OrderItem.find_discount(item) do
      total - total * discount
    else
      total
    end
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 6]
```diff
@@ -207,7 +207,77 @@ end
...
+defmodule OrderItem do
+  def find_item(id) do
+    ...
+    %{price: ..., taxes: ..., amount: ...}
+  end
+
+  def find_discount(item) do
+    ...
+  end
+
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:267]:
````suggestion
To remove this anti-pattern we can move `calculate_total_item/1` to `OrderItem`, decreasing coupling:

```elixir
defmodule OrderItem do
  def find_item(id)
  def find_discount(item)

````

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 7]
```diff
@@ -207,7 +207,77 @@ end
...
+    discount = find_discount(item)
+
+    unless is_nil(discount) do    
+      total - total * discount
+    else
+      total
+    end
+  end
+end
+```
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:280]:
````suggestion
```

This refactoring is only possible when you own both modules. If the module you are invoking belongs to another application, then it is not possible to change, and your only option is to define an additional module to augment the third-party module.
````

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 8]
```diff
@@ -207,7 +207,55 @@ end
...
+
+This refactoring is only possible when you own both modules. If the module you are invoking belongs to another application, then it is not possible to change, and your only option is to define an additional module to augment the third-party module.
```
Comment (josevalim) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:258]:
```suggestion
This refactoring is only possible when you own both modules. If the module you are invoking belongs to another application, then it is not possible to add new functions to it, and your only option is to define an additional module that augments the third-party module.
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**